inception/inception-api-annotation-experimental/src/main/java/de/tudarmstadt/ukp/inception/experimental/api/AnnotationSystemAPIImpl.java

```diff
@@ -17,402 +17,669 @@
         return null;
     }
 
-    @Override
-    public List<Annotation> getAnnotations(CAS aCas, long aProject)
+    public List<AnnotationLayer> getValidLayers(long aProjectId, List<String> disabledLayers)
     {
-        List<AnnotationListItem> items = new ArrayList<>();
-        List<AnnotationLayer> metadataLayers = annotationService
-            .listAnnotationLayer(projectService.getProject(aProject));
+        if (disabledLayers.isEmpty()) {
+            return annotationService.listAnnotationLayer(projectService.getProject(aProjectId));
+        }
 
-        for (AnnotationLayer layer : metadataLayers) {
-            if (layer.getUiName().equals("Token")) {
-                // TODO: exception later when calling renderer.getFeatures "lemma"
-                continue;
+        List<AnnotationLayer> validLayers = new ArrayList<>();
+        List<AnnotationLayer> layers = annotationService
+                .listAnnotationLayer(projectService.getProject(aProjectId));
+        for (AnnotationLayer layer : layers) {
+            if (!disabledLayers.contains(layer.getType())) {
+                validLayers.add(layer);
             }
+        }
+        return validLayers;
+    }
 
-            TypeAdapter adapter = annotationService.getAdapter(layer);
-
-            for (FeatureStructure fs : selectFS(aCas, adapter.getAnnotationType(aCas))) {
-                String labelText = "";
-                items.add(new AnnotationListItem(WebAnnoCasUtil.getAddr(fs), labelText, layer));
+    public List<Span> filterSpans(List<Span> aAnnotations, Viewport aViewport) throws Exception
+    {
+        try {
+            List<Span> filteredAnnotations = new ArrayList<>();
+            for (Span annotation : aAnnotations) {
+                for (int i = 0; i < aViewport.getViewport().size(); i++) {
+                    for (int j = aViewport.getViewport().get(i).get(0); j <= aViewport.getViewport()
+                            .get(i).get(1); j++) {
+                        if (annotation.getBegin() == j) {
+                            filteredAnnotations.add(annotation);
+                            break;
+                        }
+                    }
+                }
             }
+            return filteredAnnotations;
         }
+        catch (Exception e) {
+            LOG.error("Exception has been thrown during filterSpans()", e);
+            throw new Exception();
+        }
+    }
 
-        List<Annotation> annotations = new ArrayList<>();
+    public List<Arc> filterArcs(List<Arc> aRelations, CAS aCas, Viewport aViewport) throws Exception
+    {
+        try {
+            List<Arc> filteredArcs = new ArrayList<>();
+
+            for (Arc arc : aRelations) {
+                AnnotationFS annotation = selectAnnotationByAddr(aCas, arc.getGovernorId().getId());
+
+                for (int i = 0; i < aViewport.getViewport().size(); i++) {
+                    for (int j = aViewport.getViewport().get(i).get(0); j <= aViewport.getViewport()
```

Comment:
See comment on the spans. 

How do you handle arcs which start outside or at the edge of the viewport and end inside the viewport (or vice versa) or pass entirely over it.